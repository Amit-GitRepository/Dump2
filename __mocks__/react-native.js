var rn = require('react-native');

rn.NativeModules.RNI18n = {languages: ['en']};
rn.NativeModules.BlobModule = {BLOB_URI_SCHEME: ''};
rn.NativeModules.FirebaseDynamicLink = {
  addListener: jest.fn(),
  removeListeners: jest.fn()
};
rn.NativeModules.ReactNativeExceptionHandler = {
  setHandlerforNativeException: jest.fn()
};
jest.mock('AsyncStorage', () => ({
  getItem: jest.fn(() => Promise.resolve('{}')),
  setItem: jest.fn(() => Promise.resolve()),
  mergeItem: jest.fn(() => Promise.resolve()),
  multiGet: jest.fn(() => Promise.resolve('{}'))
}));
jest.mock('BackHandler', () => ({
  addEventListener: jest.fn()
}));
jest.mock('AppState', () => ({
  currentState: 'inactive',
  addEventListener: jest.fn()
}));
var ReactNativeART = require('react-native/Libraries/ART/ReactNativeART');

ReactNativeART.Group = rn.View;
ReactNativeART.Surface = rn.View;
ReactNativeART.Shape = rn.View;
module.exports = rn;
