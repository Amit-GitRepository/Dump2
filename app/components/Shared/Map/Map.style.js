import {colors} from '../../../themes/constants.styles';
import {shadowActiveStyle} from '../../../themes/application.styles';

export default {
  container: {
    flex: 1
  },
  map: {
    flex: 1 // added for a bugfix. Ref https://github.com/airbnb/react-native-maps/issues/842
  },
  marker: {
    width: 30,
    height: 40
  },
  buttonList: {
    position: 'absolute',
    bottom: 35,
    right: 5
  },
  buttonWrapper: {
    backgroundColor: colors.PRIMARY_ACTIONABLE,
    borderRadius: 100,
    borderWidth: 2,
    borderColor: colors.PRIMARY_BG_TEXT_CONTRAST,
    margin: 5,
    ...shadowActiveStyle
  },
  button: {
    fontSize: 55,
    borderRadius: 55,
    overflow: 'hidden',
    color: colors.PRIMARY_BG_TEXT_CONTRAST
  }
};
