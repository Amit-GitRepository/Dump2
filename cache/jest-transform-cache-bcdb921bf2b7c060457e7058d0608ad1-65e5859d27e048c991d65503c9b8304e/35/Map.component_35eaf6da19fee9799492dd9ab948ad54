Object.defineProperty(exports,"__esModule",{value:true});var cov_1u5tbcax7m=function(){var path='/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/Shared/Map/Map.component.js',hash='b573cb884fc07fcebd63d39da41ac4769fe97330',global=new Function('return this')(),gcv='__coverage__',coverageData={path:'/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/Shared/Map/Map.component.js',statementMap:{'0':{start:{line:25,column:28},end:{line:25,column:38}},'1':{start:{line:26,column:4},end:{line:28,column:5}},'2':{start:{line:27,column:6},end:{line:27,column:57}},'3':{start:{line:32,column:28},end:{line:32,column:38}},'4':{start:{line:33,column:23},end:{line:33,column:33}},'5':{start:{line:34,column:4},end:{line:36,column:5}},'6':{start:{line:35,column:6},end:{line:35,column:46}},'7':{start:{line:40,column:4},end:{line:42,column:5}},'8':{start:{line:41,column:6},end:{line:41,column:19}},'9':{start:{line:43,column:4},end:{line:43,column:16}},'10':{start:{line:46,column:30},end:{line:54,column:4}},'11':{start:{line:47,column:22},end:{line:47,column:73}},'12':{start:{line:48,column:4},end:{line:53,column:15}},'13':{start:{line:57,column:66},end:{line:57,column:76}},'14':{start:{line:58,column:4},end:{line:58,column:91}},'15':{start:{line:59,column:4},end:{line:59,column:26}},'16':{start:{line:63,column:4},end:{line:63,column:19}},'17':{start:{line:67,column:42},end:{line:67,column:52}},'18':{start:{line:68,column:38},end:{line:68,column:48}},'19':{start:{line:69,column:4},end:{line:101,column:6}},'20':{start:{line:105,column:0},end:{line:112,column:2}},'21':{start:{line:114,column:0},end:{line:122,column:2}}},fnMap:{'0':{name:'(anonymous_0)',decl:{start:{line:24,column:2},end:{line:24,column:3}},loc:{start:{line:24,column:33},end:{line:29,column:3}},line:24},'1':{name:'(anonymous_1)',decl:{start:{line:31,column:2},end:{line:31,column:3}},loc:{start:{line:31,column:40},end:{line:37,column:3}},line:31},'2':{name:'(anonymous_2)',decl:{start:{line:39,column:2},end:{line:39,column:3}},loc:{start:{line:39,column:36},end:{line:44,column:3}},line:39},'3':{name:'(anonymous_3)',decl:{start:{line:46,column:14},end:{line:46,column:15}},loc:{start:{line:46,column:30},end:{line:54,column:4}},line:46},'4':{name:'(anonymous_4)',decl:{start:{line:46,column:45},end:{line:46,column:46}},loc:{start:{line:46,column:63},end:{line:54,column:3}},line:46},'5':{name:'(anonymous_5)',decl:{start:{line:56,column:24},end:{line:56,column:25}},loc:{start:{line:56,column:30},end:{line:60,column:3}},line:56},'6':{name:'(anonymous_6)',decl:{start:{line:62,column:11},end:{line:62,column:12}},loc:{start:{line:62,column:20},end:{line:64,column:3}},line:62},'7':{name:'(anonymous_7)',decl:{start:{line:66,column:2},end:{line:66,column:3}},loc:{start:{line:66,column:12},end:{line:102,column:3}},line:66}},branchMap:{'0':{loc:{start:{line:26,column:4},end:{line:28,column:5}},type:'if',locations:[{start:{line:26,column:4},end:{line:28,column:5}},{start:{line:26,column:4},end:{line:28,column:5}}],line:26},'1':{loc:{start:{line:34,column:4},end:{line:36,column:5}},type:'if',locations:[{start:{line:34,column:4},end:{line:36,column:5}},{start:{line:34,column:4},end:{line:36,column:5}}],line:34},'2':{loc:{start:{line:40,column:4},end:{line:42,column:5}},type:'if',locations:[{start:{line:40,column:4},end:{line:42,column:5}},{start:{line:40,column:4},end:{line:42,column:5}}],line:40},'3':{loc:{start:{line:47,column:22},end:{line:47,column:73}},type:'binary-expr',locations:[{start:{line:47,column:22},end:{line:47,column:53}},{start:{line:47,column:57},end:{line:47,column:73}}],line:47},'4':{loc:{start:{line:58,column:35},end:{line:58,column:84}},type:'cond-expr',locations:[{start:{line:58,column:53},end:{line:58,column:68}},{start:{line:58,column:71},end:{line:58,column:84}}],line:58},'5':{loc:{start:{line:74,column:11},end:{line:81,column:18}},type:'cond-expr',locations:[{start:{line:75,column:14},end:{line:80,column:21}},{start:{line:81,column:14},end:{line:81,column:18}}],line:74},'6':{loc:{start:{line:82,column:11},end:{line:82,column:57}},type:'cond-expr',locations:[{start:{line:82,column:24},end:{line:82,column:50}},{start:{line:82,column:53},end:{line:82,column:57}}],line:82},'7':{loc:{start:{line:89,column:11},end:{line:94,column:18}},type:'cond-expr',locations:[{start:{line:90,column:14},end:{line:93,column:19}},{start:{line:94,column:14},end:{line:94,column:18}}],line:89}},s:{'0':0,'1':0,'2':0,'3':0,'4':0,'5':0,'6':0,'7':0,'8':0,'9':0,'10':0,'11':0,'12':0,'13':0,'14':0,'15':0,'16':0,'17':0,'18':0,'19':0,'20':0,'21':0},f:{'0':0,'1':0,'2':0,'3':0,'4':0,'5':0,'6':0,'7':0},b:{'0':[0,0],'1':[0,0],'2':[0,0],'3':[0,0],'4':[0,0],'5':[0,0],'6':[0,0],'7':[0,0]},_coverageSchema:'332fd63041d2c1bcb487cc26dd0d5f7d97098a6c'},coverage=global[gcv]||(global[gcv]={});if(coverage[path]&&coverage[path].hash===hash){return coverage[path];}coverageData.hash=hash;return coverage[path]=coverageData;}();var _createClass=function(){function defineProperties(target,props){for(var i=0;i<props.length;i++){var descriptor=props[i];descriptor.enumerable=descriptor.enumerable||false;descriptor.configurable=true;if("value"in descriptor)descriptor.writable=true;Object.defineProperty(target,descriptor.key,descriptor);}}return function(Constructor,protoProps,staticProps){if(protoProps)defineProperties(Constructor.prototype,protoProps);if(staticProps)defineProperties(Constructor,staticProps);return Constructor;};}();var _Icon=require('../Icon/Icon.component');var _Icon2=_interopRequireDefault(_Icon);
var _isEqual=require('lodash/isEqual');var _isEqual2=_interopRequireDefault(_isEqual);
var _map=require('../../../config/map.config');var _map2=_interopRequireDefault(_map);
var _reactNativeMapMarkerclustering=require('react-native-map-markerclustering');var _reactNativeMapMarkerclustering2=_interopRequireDefault(_reactNativeMapMarkerclustering);
var _noop=require('lodash/noop');var _noop2=_interopRequireDefault(_noop);
var _pinLocation=require('../../../assets/pinLocation.png');var _pinLocation2=_interopRequireDefault(_pinLocation);
var _propTypes=require('prop-types');var _propTypes2=_interopRequireDefault(_propTypes);
var _react=require('react');var _react2=_interopRequireDefault(_react);
var _shopMarker=require('../../../assets/shopMarker.png');var _shopMarker2=_interopRequireDefault(_shopMarker);
var _Map=require('./Map.style');var _Map2=_interopRequireDefault(_Map);
var _Touchable=require('../Touchable/Touchable.component');var _Touchable2=_interopRequireDefault(_Touchable);
var _trueShopMapViewPage=require('../../../config/testid/trueShopMapViewPage');var _trueShopMapViewPage2=_interopRequireDefault(_trueShopMapViewPage);
var _constants=require('../../../themes/constants.styles');
var _reactNative=require('react-native');
var _reactNativeMaps=require('react-native-maps');function _interopRequireDefault(obj){return obj&&obj.__esModule?obj:{default:obj};}function _classCallCheck(instance,Constructor){if(!(instance instanceof Constructor)){throw new TypeError("Cannot call a class as a function");}}function _possibleConstructorReturn(self,call){if(!self){throw new ReferenceError("this hasn't been initialised - super() hasn't been called");}return call&&(typeof call==="object"||typeof call==="function")?call:self;}function _inherits(subClass,superClass){if(typeof superClass!=="function"&&superClass!==null){throw new TypeError("Super expression must either be null or a function, not "+typeof superClass);}subClass.prototype=Object.create(superClass&&superClass.prototype,{constructor:{value:subClass,enumerable:false,writable:true,configurable:true}});if(superClass)Object.setPrototypeOf?Object.setPrototypeOf(subClass,superClass):subClass.__proto__=superClass;}var

Map=function(_Component){_inherits(Map,_Component);function Map(){var _ref;var _temp,_this,_ret;_classCallCheck(this,Map);for(var _len=arguments.length,args=Array(_len),_key=0;_key<_len;_key++){args[_key]=arguments[_key];}return _ret=(_temp=(_this=_possibleConstructorReturn(this,(_ref=Map.__proto__||Object.getPrototypeOf(Map)).call.apply(_ref,[this].concat(args))),_this),_this.

state={
markerId:1,
initialRegion:_this.props.initialRegion},_this.
























addMarker=function(markerList){cov_1u5tbcax7m.f[3]++;cov_1u5tbcax7m.s[10]++;return markerList.map(function(store,index){cov_1u5tbcax7m.f[4]++;
var storeName=(cov_1u5tbcax7m.s[11]++,(cov_1u5tbcax7m.b[3][0]++,store.name[_this.props.language])||(cov_1u5tbcax7m.b[3][1]++,store.name['en']));cov_1u5tbcax7m.s[12]++;
return _react2.default.createElement(_reactNativeMaps.Marker,{key:index,coordinate:store,title:storeName,onCalloutPress:_this.props.onCalloutPress(store)},
_react2.default.createElement(_reactNative.Image,{
style:_Map2.default.marker,
source:_shopMarker2.default}));


});},_this.

showCurrentLocation=function(){cov_1u5tbcax7m.f[5]++;var _ref2=(cov_1u5tbcax7m.s[13]++,
_this.props),currentLocation=_ref2.currentLocation,initialRegion=_ref2.initialRegion,onPressCurrLocation=_ref2.onPressCurrLocation;cov_1u5tbcax7m.s[14]++;
_this.map._root.animateToRegion(currentLocation?(cov_1u5tbcax7m.b[4][0]++,currentLocation):(cov_1u5tbcax7m.b[4][1]++,initialRegion),500);cov_1u5tbcax7m.s[15]++;
onPressCurrLocation();
},_this.

setRef=function(map){cov_1u5tbcax7m.f[6]++;cov_1u5tbcax7m.s[16]++;
_this.map=map;
},_temp),_possibleConstructorReturn(_this,_ret);}_createClass(Map,[{key:'componentDidUpdate',value:function componentDidUpdate(prevProps){cov_1u5tbcax7m.f[0]++;var _ref3=(cov_1u5tbcax7m.s[0]++,this.props),initialRegion=_ref3.initialRegion;cov_1u5tbcax7m.s[1]++;if(initialRegion!==prevProps.initialRegion){cov_1u5tbcax7m.b[0][0]++;cov_1u5tbcax7m.s[2]++;this.map._root.animateToRegion(initialRegion,500);}else{cov_1u5tbcax7m.b[0][1]++;}}},{key:'componentWillReceiveProps',value:function componentWillReceiveProps(nextProps){cov_1u5tbcax7m.f[1]++;var _ref4=(cov_1u5tbcax7m.s[3]++,this.props),initialRegion=_ref4.initialRegion;var _ref5=(cov_1u5tbcax7m.s[4]++,this.state),markerId=_ref5.markerId;cov_1u5tbcax7m.s[5]++;if(initialRegion!==nextProps.initialRegion){cov_1u5tbcax7m.b[1][0]++;cov_1u5tbcax7m.s[6]++;this.setState({markerId:markerId+1});}else{cov_1u5tbcax7m.b[1][1]++;}}},{key:'shouldComponentUpdate',value:function shouldComponentUpdate(nextProps){cov_1u5tbcax7m.f[2]++;cov_1u5tbcax7m.s[7]++;if((0,_isEqual2.default)(nextProps,this.props)){cov_1u5tbcax7m.b[2][0]++;cov_1u5tbcax7m.s[8]++;return false;}else{cov_1u5tbcax7m.b[2][1]++;}cov_1u5tbcax7m.s[9]++;return true;}},{key:'render',value:function render()

{cov_1u5tbcax7m.f[7]++;var _ref6=(cov_1u5tbcax7m.s[17]++,
this.props),markerList=_ref6.markerList,currentLocation=_ref6.currentLocation;var _ref7=(cov_1u5tbcax7m.s[18]++,
this.state),markerId=_ref7.markerId,initialRegion=_ref7.initialRegion;cov_1u5tbcax7m.s[19]++;
return(
_react2.default.createElement(_reactNative.View,{style:_Map2.default.container},
_react2.default.createElement(_reactNativeMapMarkerclustering2.default,{style:_Map2.default.map,ref:this.setRef,region:initialRegion,
clusterColor:_constants.colors.MAPCLUSTER,clusterTextColor:_constants.colors.PRIMARY_BG_TEXT_CONTRAST,
clusterBorderColor:_constants.colors.PRIMARY_BG_TEXT_CONTRAST},
currentLocation?(cov_1u5tbcax7m.b[5][0]++,
_react2.default.createElement(_reactNativeMaps.Marker,{key:'currentMarker_'+markerId,coordinate:currentLocation,disableClustering:true},
_react2.default.createElement(_reactNative.Image,{
style:_Map2.default.marker,
source:_pinLocation2.default}))):(cov_1u5tbcax7m.b[5][1]++,


null),
markerList?(cov_1u5tbcax7m.b[6][0]++,this.addMarker(markerList)):(cov_1u5tbcax7m.b[6][1]++,null)),

_react2.default.createElement(_reactNative.View,{style:_Map2.default.buttonList},
_react2.default.createElement(_reactNative.View,{style:_Map2.default.buttonWrapper,elevation:4},_react2.default.createElement(_Touchable2.default,{onPress:this.props.onSearchPress},
_react2.default.createElement(_Icon2.default,{name:'map-search',style:_Map2.default.button,accessibilityLabel:_trueShopMapViewPage2.default.ICON_SEARCHSTORE,testID:_trueShopMapViewPage2.default.ICON_SEARCHSTORE}))),


currentLocation?(cov_1u5tbcax7m.b[7][0]++,
_react2.default.createElement(_reactNative.View,{style:_Map2.default.buttonWrapper,elevation:4},_react2.default.createElement(_Touchable2.default,{onPress:this.showCurrentLocation},
_react2.default.createElement(_Icon2.default,{name:'current-position',style:_Map2.default.button,accessibilityLabel:_trueShopMapViewPage2.default.ICON_GOTOCURRENTLOCATION,testID:_trueShopMapViewPage2.default.ICON_GOTOCURRENTLOCATION})))):(cov_1u5tbcax7m.b[7][1]++,


null))));







}}]);return Map;}(_react.Component);cov_1u5tbcax7m.s[20]++;


Map.defaultProps={
markerList:[],
initialRegion:_map2.default.initialRegion,
currentLocation:null,
onPressCurrLocation:_noop2.default,
onCalloutPress:_noop2.default,
onSearchPress:_noop2.default};cov_1u5tbcax7m.s[21]++;


Map.propTypes={
markerList:_propTypes2.default.array,
initialRegion:_propTypes2.default.object,
currentLocation:_propTypes2.default.object,
onPressCurrLocation:_propTypes2.default.func,
onCalloutPress:_propTypes2.default.func,
onSearchPress:_propTypes2.default.func,
language:_propTypes2.default.string};exports.default=


Map;