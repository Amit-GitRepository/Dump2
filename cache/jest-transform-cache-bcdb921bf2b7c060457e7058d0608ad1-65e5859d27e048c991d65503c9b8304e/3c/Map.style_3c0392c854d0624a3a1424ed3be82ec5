Object.defineProperty(exports,"__esModule",{value:true});var cov_2qq4yddzbt=function(){var path='/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/Shared/Map/Map.style.js',hash='987d798e89c3a3a88b97debcd23f9fde1b17eed1',global=new Function('return this')(),gcv='__coverage__',coverageData={path:'/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/Shared/Map/Map.style.js',statementMap:{},fnMap:{},branchMap:{},s:{},f:{},b:{},_coverageSchema:'332fd63041d2c1bcb487cc26dd0d5f7d97098a6c'},coverage=global[gcv]||(global[gcv]={});if(coverage[path]&&coverage[path].hash===hash){return coverage[path];}coverageData.hash=hash;return coverage[path]=coverageData;}();var _constants=require('../../../themes/constants.styles');exports.default=

{
container:{
flex:1},

map:{
flex:1},

buttonList:{
position:'absolute',
bottom:35,
right:5},

buttonWrapper:{
backgroundColor:_constants.colors.PRIMARY_ACTIONABLE,
borderRadius:100,
borderWidth:2,
borderColor:_constants.colors.PRIMARY_BG_TEXT_CONTRAST,
overflow:'hidden',
margin:5},

button:{
fontSize:55,
color:_constants.colors.PRIMARY_BG_TEXT_CONTRAST}};