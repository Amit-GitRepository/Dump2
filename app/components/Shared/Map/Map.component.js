import Icon from '../Icon/Icon.component';
import isEqual from 'lodash/isEqual';
import mapConfig from '../../../config/map.config';
import MapView from 'react-native-map-markerclustering';
import noop from 'lodash/noop';
import pinLocation from '../../../assets/pinLocation.png';
import PropTypes from 'prop-types';
import React, {Component} from 'react';
import shopMarker from '../../../assets/shopMarker.png';
import styles from './Map.style';
import Touchable from '../Touchable/Touchable.component';
import trueShopMapViewTestIDs from '../../../config/testid/trueShopMapViewPage';
import {colors} from '../../../themes/constants.styles';
import {Image, View} from 'react-native';
import {Marker} from 'react-native-maps';

class Map extends Component {

  state = {
    markerId: 1,
    initialRegion: this.props.initialRegion
  }

  componentDidUpdate (prevProps) {
    const {initialRegion} = this.props;
    if (initialRegion !== prevProps.initialRegion) {
      this.map._root.animateToRegion(initialRegion, 500);
    }
  }

  componentWillReceiveProps (nextProps) {
    const {initialRegion} = this.props;
    const {markerId} = this.state;
    if (initialRegion !== nextProps.initialRegion) {
      this.setState({markerId: markerId + 1});
    }
  }

  shouldComponentUpdate (nextProps) {
    if (isEqual(nextProps, this.props)) {
      return false;
    }
    return true;
  }

  addMarker = (markerList) => markerList.map((store, index) => {
    const storeName = store.name[this.props.language] || store.name['en'];
    return (<Marker key={index} coordinate={store} title={storeName} onCalloutPress={this.props.onCalloutPress(store)}>
      <Image
        style={styles.marker}
        source={shopMarker}
      />
    </Marker>);
  })

  showCurrentLocation = () => {
    const {currentLocation, initialRegion, onPressCurrLocation} = this.props;
    this.map._root.animateToRegion(currentLocation ? currentLocation : initialRegion, 500);
    onPressCurrLocation();
  }

  setRef = (map) => {
    this.map = map;
  }

  render () {
    const {markerList, currentLocation} = this.props;
    const {markerId, initialRegion} = this.state;
    return (
      <View style={styles.container}>
        <MapView style={styles.map} ref={this.setRef} region={initialRegion}
          clusterColor={colors.MAPCLUSTER} clusterTextColor={colors.PRIMARY_BG_TEXT_CONTRAST}
          clusterBorderColor={colors.PRIMARY_BG_TEXT_CONTRAST}>
          {currentLocation
            ? <Marker key={`currentMarker_${markerId}`} coordinate={currentLocation} disableClustering={true}>
              <Image
                style={styles.marker}
                source={pinLocation}
              />
            </Marker>
            : null}
          {markerList ? this.addMarker(markerList) : null}
        </MapView>
        <View style={styles.buttonList}>
          <View style={styles.buttonWrapper} elevation={4}><Touchable onPress={this.props.onSearchPress}>
            <Icon name='map-search' style={styles.button} accessibilityLabel={trueShopMapViewTestIDs.ICON__SEARCHSTORE} testID={trueShopMapViewTestIDs.ICON__SEARCHSTORE}/>
          </Touchable>
          </View>
          {currentLocation
            ? <View style={styles.buttonWrapper} elevation={4}><Touchable onPress={this.showCurrentLocation}>
              <Icon name='current-position' style={styles.button} accessibilityLabel={trueShopMapViewTestIDs.ICON__GOTOCURRENTLOCATION} testID={trueShopMapViewTestIDs.ICON__GOTOCURRENTLOCATION}/>
            </Touchable>
            </View>
            : null}
          {/* The remark is currently commented out as clustering of markers is still not decided */}
          {/* <Touchable style={styles.buttonWrapper} onPress={null}>
            <Icon name='remark' style={styles.button}/>
          </Touchable> */}
        </View>
      </View>
    );
  }
}

Map.defaultProps = {
  markerList: [],
  initialRegion: mapConfig.initialRegion,
  currentLocation: null,
  onPressCurrLocation: noop,
  onCalloutPress: noop,
  onSearchPress: noop
};

Map.propTypes = {
  markerList: PropTypes.array,
  initialRegion: PropTypes.object,
  currentLocation: PropTypes.object,
  onPressCurrLocation: PropTypes.func,
  onCalloutPress: PropTypes.func,
  onSearchPress: PropTypes.func,
  language: PropTypes.string
};

export default Map;
