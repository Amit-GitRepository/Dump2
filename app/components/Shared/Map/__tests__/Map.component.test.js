import Map from '../Map.component';
import React from 'react';
import ReactSixteenAdapter from 'enzyme/build/adapters/ReactSixteenAdapter';
import renderer from 'react-test-renderer'; // Enzyme adapter for React 16
import shopMarker from '../../../../assets/shopMarker.png';
import {configure, shallow} from 'enzyme';
import {Image} from 'react-native';
import {Marker} from 'react-native-maps';

configure({adapter: new ReactSixteenAdapter()});
const initialRegion = {
  latitude: 12.9142059,
  longitude: 77.6820747,
  latitudeDelta: 0.012,
  longitudeDelta: 0.034
};
describe('Map component', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Map initialRegion={initialRegion}/>);
    const component = renderer.create(<Map/>).toJSON();
    expect(component).toMatchSnapshot();
    expect(wrapper).toBeDefined();
  });

  it('should render markers inside map if markerList is passed', () => {
    const markerList = [
      {
        latitude: 12.9142059,
        longitude: 77.6820747,
        name: {
          'en': 'some shop',
          'th': 'some shop (th)'
        }
      }
    ];
    const wrapper = shallow(<Map initialRegion={initialRegion} markerList={markerList} language={'en'}/>);
    const component = renderer.create(<Map initialRegion={initialRegion} markerList={markerList} language={'en'}/>).toJSON();
    expect(component).toMatchSnapshot();
    expect(wrapper).toBeDefined();
  });

  it('should render map with the initial region if initialRegion is passed', () => {
    const wrapper = shallow(<Map initialRegion={initialRegion}/>);
    const component = renderer.create(<Map initialRegion={initialRegion}/>).toJSON();
    expect(component).toMatchSnapshot();
    expect(wrapper).toBeDefined();
  });

  it('addMarker: should return Marker component when marker array is passed', () => {
    const markerList = [
      {
        latitude: 12.9142059,
        longitude: 77.6820747,
        name: {
          'en': 'some shop',
          'th': 'some shop (th)'
        }
      }
    ];
    const wrapper = shallow(<Map initialRegion={initialRegion} markerList={markerList} language={'en'} onCalloutPress={jest.fn()}/>);
    const marker = [<Marker key={0}
      coordinate={{latitude: 12.9142059, longitude: 77.6820747, name: {'en': 'some shop', 'th': 'some shop (th)'}}} title={'some shop'}>
      <Image
        style={{width: 30, height: 40}}
        source={shopMarker}
      />
    </Marker>];
    expect(wrapper.instance().addMarker(markerList)).toEqual(marker);
  });

  it('addMarker: should return empty array when empty markerList array is passed', () => {
    const markerList = [];
    const wrapper = shallow(<Map initialRegion={initialRegion} markerList={markerList}/>);
    const marker = [];
    expect(wrapper.instance().addMarker(markerList)).toEqual(marker);
  });

  it('showCurrentLocation: should update map region to current location', () => {
    const currentLocation = {
      latitude: 10.12345,
      longitude: 70.123457,
      latitudeDelta: 0.098,
      longitudeDelta: 0.076
    };

    const map = {
      _root: {animateToRegion: jest.fn()}
    };

    const wrapper = shallow(<Map initialRegion={initialRegion} currentLocation={currentLocation} onPressCurrLocation={jest.fn()}/>);
    wrapper.instance().setRef(map);
    wrapper.instance().showCurrentLocation();
    expect(map._root.animateToRegion).toHaveBeenCalled();
  });

  it('componentDidUpdate: should update map region to the initial region prop', () => {
    const currentLocation = {
      latitude: 10.12345,
      longitude: 70.123457,
      latitudeDelta: 0.098,
      longitudeDelta: 0.076
    };

    const map = {
      _root: {animateToRegion: jest.fn()}
    };

    const wrapper = shallow(<Map initialRegion={initialRegion} currentLocation={currentLocation} onPressCurrLocation={jest.fn()}/>);
    wrapper.instance().setRef(map);
    wrapper.instance().componentDidUpdate({initialRegion: currentLocation});
    expect(map._root.animateToRegion).toHaveBeenCalled();
  });
});
