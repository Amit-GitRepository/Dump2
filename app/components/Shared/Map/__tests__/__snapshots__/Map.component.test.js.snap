// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map component renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "bottom": 35,
        "position": "absolute",
        "right": 5,
      }
    }
  >
    <View
      elevation={4}
      style={
        Object {
          "backgroundColor": "#F93E60",
          "borderColor": "#FFFFFF",
          "borderRadius": 100,
          "borderWidth": 2,
          "margin": 5,
          "shadowColor": "#303C46",
          "shadowOffset": Object {
            "height": 2,
            "width": 1,
          },
          "shadowOpacity": 0.3,
          "shadowRadius": 1,
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            accessibilityLabel="trueShopMapView_icon_searchStore"
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 12,
                },
                Object {
                  "borderRadius": 55,
                  "color": "#FFFFFF",
                  "fontSize": 55,
                  "overflow": "hidden",
                },
                Object {
                  "fontFamily": "iservice",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
            testID="trueShopMapView_icon_searchStore"
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`Map component should render map with the initial region if initialRegion is passed 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "bottom": 35,
        "position": "absolute",
        "right": 5,
      }
    }
  >
    <View
      elevation={4}
      style={
        Object {
          "backgroundColor": "#F93E60",
          "borderColor": "#FFFFFF",
          "borderRadius": 100,
          "borderWidth": 2,
          "margin": 5,
          "shadowColor": "#303C46",
          "shadowOffset": Object {
            "height": 2,
            "width": 1,
          },
          "shadowOpacity": 0.3,
          "shadowRadius": 1,
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            accessibilityLabel="trueShopMapView_icon_searchStore"
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 12,
                },
                Object {
                  "borderRadius": 55,
                  "color": "#FFFFFF",
                  "fontSize": 55,
                  "overflow": "hidden",
                },
                Object {
                  "fontFamily": "iservice",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
            testID="trueShopMapView_icon_searchStore"
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`Map component should render markers inside map if markerList is passed 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "bottom": 35,
        "position": "absolute",
        "right": 5,
      }
    }
  >
    <View
      elevation={4}
      style={
        Object {
          "backgroundColor": "#F93E60",
          "borderColor": "#FFFFFF",
          "borderRadius": 100,
          "borderWidth": 2,
          "margin": 5,
          "shadowColor": "#303C46",
          "shadowOffset": Object {
            "height": 2,
            "width": 1,
          },
          "shadowOpacity": 0.3,
          "shadowRadius": 1,
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            accessibilityLabel="trueShopMapView_icon_searchStore"
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 12,
                },
                Object {
                  "borderRadius": 55,
                  "color": "#FFFFFF",
                  "fontSize": 55,
                  "overflow": "hidden",
                },
                Object {
                  "fontFamily": "iservice",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
            testID="trueShopMapView_icon_searchStore"
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
